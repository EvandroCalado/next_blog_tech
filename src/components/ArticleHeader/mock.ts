export default {
  id: '3',
  title: 'Desenvolvimento de aplicativos web escaláveis com Next.js',
  excerpt:
    'Se você está procurando um framework poderoso para desenvolver aplicativos web escaláveis, o Next.js pode ser a escolha certa para você. O Next.js é um framework baseado em React que oferece recursos avançados, como renderização no lado do servidor e roteamento dinâmico, para ajudá-lo a criar aplicativos web modernos e de alto desempenho. Neste post, exploraremos as principais características do Next.js e veremos como usá-lo para construir aplicativos web escaláveis.',
  createdAt: '2023-03-27T17:22:39.749Z',
  cover:
    'https://res.cloudinary.com/dnhsdenji/image/upload/v1679929576/twitter_card_f0df8b6a20.png',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
