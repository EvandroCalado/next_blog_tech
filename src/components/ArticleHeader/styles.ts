import styled, { css } from 'styled-components';
import { Title } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacings.xlarge};
    margin-bottom: ${theme.spacings.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${Title} {
      margin: 0;
      margin-bottom: ${theme.spacings.medium};

      @media ${theme.media.small} {
        font-size: ${theme.font.sizes.medium};
      }
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacings.medium} 0;
    font-size: ${theme.font.sizes.medium};

    @media ${theme.media.small} {
      font-size: ${theme.font.sizes.small};
    }
  `}
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    width: 100%;
    display: block;
    margin-bottom: ${theme.spacings.medium};
  `}
`;
