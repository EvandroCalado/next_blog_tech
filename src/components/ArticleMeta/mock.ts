export default {
  createdAt: '2023-03-27T17:22:39.749Z',
  author: {
    displayName: 'Evandro Calado',
    slug: 'evandro-calado',
  },
  categories: [
    {
      attributes: {
        displayName: 'React',
        slug: 'react',
      },
    },
    {
      attributes: {
        displayName: 'Next',
        slug: 'next',
      },
    },
  ],
};
