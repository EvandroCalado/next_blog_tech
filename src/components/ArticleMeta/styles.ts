import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.darkerGray};
    font-style: italic;

    .categories span:not(:last-child)::after {
      content: ', ';
    }

    span:first-child {
      margin-right: 0.5rem;
    }

    a {
      color: ${theme.colors.secondary};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
