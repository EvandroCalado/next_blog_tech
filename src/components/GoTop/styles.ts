import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    background-color: ${theme.colors.primary};
    color: ${theme.colors.white};
    position: fixed;
    bottom: 3rem;
    right: 5rem;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    z-index: 6;
    text-decoration: none;
    transition: all 300ms ease-in-out;

    &:hover {
      color: ${theme.colors.secondary};
    }
  `}
`;
