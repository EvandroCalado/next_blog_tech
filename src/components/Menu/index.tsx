import * as Styled from './styles';
import {
  Menu as MenuIcon,
  Close as CloseIcon,
} from '@styled-icons/material-outlined';
import LogoLink from '../LogoLink';
import MenuLink from '../MenuLink';
import { useState } from 'react';

export type MenuPropsLinks = {
  id: string;
  link: string;
  newTab?: boolean;
  text: string;
};

export type MenuProps = {
  links: MenuPropsLinks[];
  blogName: string;
  logo: string;
};
const Menu = ({ links = [], blogName, logo }: MenuProps) => {
  const [menuVisible, setMenuVisible] = useState(false);

  const handleOpenCloseMenu = (event: React.MouseEvent) => {
    event.preventDefault();
    setMenuVisible(!menuVisible);
  };

  return (
    <>
      <Styled.OpenClose
        onClick={handleOpenCloseMenu}
        menuVisible={menuVisible}
        href="#"
        aria-label="Open or close menu"
        title="Open or close menu"
      >
        {menuVisible && <CloseIcon aria-label="Close menu" />}
        {!menuVisible && <MenuIcon aria-label="Open menu" />}
      </Styled.OpenClose>

      <Styled.Wrapper menuVisible={menuVisible} aria-hidden={!menuVisible}>
        <Styled.Nav>
          <Styled.Logo>
            <LogoLink link="/" text={blogName} image={logo} />
          </Styled.Logo>
          {links.map((link) => (
            <MenuLink key={link.id} link={link.link} newTab={link.newTab}>
              {link.text}
            </MenuLink>
          ))}
        </Styled.Nav>
      </Styled.Wrapper>
    </>
  );
};

export default Menu;
