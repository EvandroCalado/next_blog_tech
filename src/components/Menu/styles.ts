import styled, { css, DefaultTheme } from 'styled-components';
import { Title } from '../Heading/styles';

type MenuBehaviorProps = {
  menuVisible: boolean;
};

const wrapperChanger = (
  menuVisible: MenuBehaviorProps['menuVisible'],
  theme: DefaultTheme,
) => css`
  left: ${menuVisible ? '0' : '-30rem'};

  @media ${theme.media.medium} {
    left: ${menuVisible ? '0' : '-32rem'};
  }
`;

export const Wrapper = styled.div<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    background-color: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    flex-direction: column;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    overflow-y: auto;
    transition: all 300ms ease-in-out;
    ${wrapperChanger(menuVisible, theme)}

    &::-webkit-scrollbar {
      display: none;
    }
  `}
`;

export const Nav = styled.nav`
  margin: auto;
  width: 100%;
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    ${Title} {
      display: flex;
      justify-content: center;
      margin: ${theme.spacings.xlarge} 0;

      img {
        border: 0.3rem solid ${theme.colors.secondary};
      }
    }
  `}
`;

const buttonChanger = (
  menuVisible: MenuBehaviorProps['menuVisible'],
  theme: DefaultTheme,
) => css`
  left: ${menuVisible ? '26rem' : '1rem'};
  color: ${menuVisible ? theme.colors.secondary : theme.colors.white};

  @media ${theme.media.medium} {
    left: ${menuVisible ? '26rem' : '0'};
  }
`;

export const OpenClose = styled.a<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    color: ${theme.colors.white};
    background-color: ${theme.colors.primary};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    left: 26rem;
    transition: all 300ms ease-in-out;
    ${buttonChanger(menuVisible, theme)}
  `}
`;
