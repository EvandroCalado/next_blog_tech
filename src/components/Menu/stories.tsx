import Menu, { MenuProps } from '.';
import { Meta, Story } from '@storybook/react/types-6-0';

import mock from './mock';

export default {
  title: 'Menu',
  component: Menu,
  args: mock,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta;

export const Template: Story<MenuProps> = (args) => (
  <div style={{ maxWidth: '96rem', margin: '5rem auto', padding: '0 3.2rem' }}>
    <Menu {...args} />
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer nec odio.
      Praesent libero. Sed cursus ante dapibus diam. Sed nisi. Nulla quis sem at
      nibh elementum imperdiet. Duis sagittis ipsum.{' '}
      <b>Lorem ipsum dolor sit amet, consectetur adipiscing elit</b>. Praesent
      mauris. Fusce nec tellus sed augue semper porta. Mauris massa. Vestibulum
      lacinia arcu eget nulla. Class aptent taciti sociosqu ad litora torquent
      per conubia nostra, per inceptos himenaeos.{' '}
    </p>

    <br />

    <p>
      Curabitur sodales ligula in libero. Sed dignissim lacinia nunc. Curabitur
      tortor. Pellentesque nibh. Aenean quam. In scelerisque sem at dolor.
      Maecenas mattis. Sed convallis tristique sem.{' '}
      <i>Lorem ipsum dolor sit amet, consectetur adipiscing elit</i>. Proin ut
      ligula vel nunc egestas porttitor. Morbi lectus risus, iaculis vel,
      suscipit quis, luctus non, massa. Fusce ac turpis quis ligula lacinia
      aliquet. Mauris ipsum. <b>Curabitur sodales ligula in libero</b>. Nulla
      metus metus, ullamcorper vel, tincidunt sed, euismod in, nibh. Quisque
      volutpat condimentum velit.{' '}
    </p>

    <br />

    <p>
      Class aptent taciti sociosqu ad litora torquent per conubia nostra, per
      inceptos himenaeos. Nam nec ante. <i>Curabitur tortor</i>. Sed lacinia,
      urna non tincidunt mattis, tortor neque adipiscing diam, a cursus ipsum
      ante quis turpis. Nulla facilisi. Ut fringilla. Suspendisse potenti. Nunc
      feugiat mi a tellus consequat imperdiet. Vestibulum sapien. Proin quam.
      Etiam ultrices. Suspendisse in justo eu magna luctus suscipit. Sed lectus.{' '}
    </p>

    <br />

    <p>
      <i>Sed convallis tristique sem</i>. Integer euismod lacus luctus magna.{' '}
      <b>
        Class aptent taciti sociosqu ad litora torquent per conubia nostra, per
        inceptos himenaeos
      </b>
      . Quisque cursus, metus vitae pharetra auctor, sem massa mattis sem, at
      interdum magna augue eget diam.{' '}
      <b>Nunc feugiat mi a tellus consequat imperdiet</b>. Vestibulum ante ipsum
      primis in faucibus orci luctus et ultrices posuere cubilia Curae; Morbi
      lacinia molestie dui. Praesent blandit dolor. Sed non quam. In vel mi sit
      amet augue congue elementum. Morbi in ipsum sit amet pede facilisis
      laoreet. <b>Suspendisse in justo eu magna luctus suscipit</b>. Donec lacus
      nunc, viverra nec, blandit vel, egestas et, augue. Vestibulum tincidunt
      malesuada tellus. Ut ultrices ultrices enim.{' '}
      <b>Suspendisse in justo eu magna luctus suscipit</b>. Curabitur sit amet
      mauris. Morbi in dui quis est pulvinar ullamcorper.{' '}
    </p>

    <br />

    <p>
      Nulla facilisi. Integer lacinia sollicitudin massa. Cras metus. Sed
      aliquet risus a tortor.{' '}
      <i>Suspendisse in justo eu magna luctus suscipit</i>. Integer id quam.
      Morbi mi. Quisque nisl felis, venenatis tristique, dignissim in, ultrices
      sit amet, augue. Proin sodales libero eget ante. Nulla quam. Aenean
      laoreet. <b>Nulla facilisi</b>. Vestibulum nisi lectus, commodo ac,
      facilisis ac, ultricies eu, pede. Ut orci risus, accumsan porttitor,
      cursus quis, aliquet eget, justo. Sed pretium blandit orci.{' '}
    </p>
  </div>
);
