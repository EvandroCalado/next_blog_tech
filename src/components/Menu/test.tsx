import Menu from '.';
import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

describe('<Menu />', () => {
  it('should render button link', () => {
    renderTheme(<Menu {...mock} links={undefined} />);

    const buttonLink = screen.getByRole('link', { name: 'Open or close menu' });
    const openMenuIcon = screen.getByLabelText('Open menu');

    expect(buttonLink).toBeInTheDocument();
    expect(openMenuIcon).toBeInTheDocument();

    expect(screen.queryByLabelText('Close menu')).not.toBeInTheDocument();
    expect(screen.queryByRole('navigation')).not.toBeInTheDocument();
  });

  it('should open/close menu on button click', () => {
    renderTheme(<Menu {...mock} />);

    const buttonLink = screen.getByRole('link', { name: 'Open or close menu' });
    fireEvent.click(buttonLink);

    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();
    expect(screen.queryByLabelText('Open menu')).not.toBeInTheDocument();
    expect(screen.getByRole('navigation')).toBeInTheDocument();
    expect(
      screen.getByRole('navigation').querySelectorAll('a:not([href="/"])'),
    ).toHaveLength(mock.links.length);
    expect(screen.getByRole('heading', { name: 'Evandro Calado' }));
    expect(screen.getByRole('img', { name: 'Evandro Calado' }));

    fireEvent.click(buttonLink);

    expect(screen.queryByLabelText('Close menu')).not.toBeInTheDocument();
    expect(screen.queryByRole('navigation')).not.toBeInTheDocument();
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<Menu {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
