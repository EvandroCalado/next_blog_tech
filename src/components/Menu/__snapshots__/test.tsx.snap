// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should match snapshot 1`] = `
.c6 {
  color: #000;
  line-height: 1.2;
  font-size: 2.4rem;
  font-weight: 900;
  text-transform: uppercase;
  text-align: center;
}

.c2 {
  background-color: #000;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  overflow-y: auto;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: -30rem;
}

.c2::-webkit-scrollbar {
  display: none;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 4rem 0;
}

.c4 .c5 img {
  border: 0.3rem solid #dc143c;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #fff;
  background-color: #000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: 1rem;
  color: #fff;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 a {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 a > img {
  width: 15rem;
  border-radius: 50%;
}

.c8 a {
  display: block;
  color: #fff;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0 solid #000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8 a:hover {
  border-right: 0.5rem solid #dc143c;
  color: #dc143c;
}

@media (max-width:768px) {
  .c2 {
    left: -32rem;
  }
}

@media (max-width:768px) {
  .c0 {
    left: 0;
  }
}

<div>
  <a
    aria-label="Open or close menu"
    class="c0"
    href="#"
    title="Open or close menu"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </a>
  <div
    aria-hidden="true"
    class="c2"
  >
    <nav
      class="c3"
    >
      <div
        class="c4"
      >
        <h1
          class="c5 c6"
        >
          <div
            class="c7"
          >
            <a
              href="/"
              target="_self"
            >
              <img
                alt="Evandro Calado"
                src="https://res.cloudinary.com/dnhsdenji/image/upload/v1679924887/110628201_ae0753864b.jpg"
              />
            </a>
          </div>
        </h1>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/1"
          target="_self"
        >
          My jobs
        </a>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/2"
          target="_self"
        >
          About me
        </a>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/3"
          target="_self"
        >
          Services
        </a>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/4"
          target="_self"
        >
          External Link
        </a>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/5"
          target="_self"
        >
          JavaScript
        </a>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/6"
          target="_self"
        >
          Python
        </a>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/7"
          target="_self"
        >
          Consectetur
        </a>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/8"
          target="_self"
        >
          Text test
        </a>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/9"
          target="_self"
        >
          Pac-man
        </a>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/10"
          target="_self"
        >
          Star Wars
        </a>
      </div>
      <div
        class="c8"
      >
        <a
          href="/home/11"
          target="_self"
        >
          A menu
        </a>
      </div>
    </nav>
  </div>
</div>
`;
